block/retriever.go

```diff
@@ -36,6 +37,22 @@ func (m *Manager) ApplyBatchFromSL(slBatch *settlement.Batch) error {
 				continue
 			}
 
+			if block.GetRevision() != m.State.GetRevision() {
+
+				rollapp, err := m.SLClient.GetRollapp()
+				if err != nil {
+					return err
+				}
+				if m.shouldStopNode(rollapp, m.State.GetRevision()) {
+					err = m.createInstruction(rollapp)
+					if err != nil {
+						return err
+					}
+					m.freezeNode(context.Background(), fmt.Errorf("syncing to fork height. please restart the node. local_block_height: %d rollapp_revision_start_height: %d local_revision: %d rollapp_revision: %d", m.State.Height(), rollapp.RevisionStartHeight, m.State.GetRevision(), rollapp.Revision))
+					return nil
+				}
```

Comment:
same snippet as `m.checkForkUpdate()`

```suggestion
			m.checkForkUpdate()
```

Reply:
replaced by checkForkUpdate, adding stop message in the func.